packages/web-components/fast-foundation/src/avatar/avatar.ts

```diff
@@ -0,0 +1,89 @@
+import { attr, SyntheticViewTemplate } from "@microsoft/fast-element";
+import { FoundationElement, FoundationElementDefinition } from "../foundation-element";
+
+type AvatarShape = "circle" | "square";
+
+/**
+ * Avatar configuration options
+ * @public
+ */
+export type AvatarOptions = FoundationElementDefinition & {
+    media?: string | SyntheticViewTemplate;
+};
+
+/**
+ * An Avatar Custom HTML Element
+ *
+ * @public
+ */
+export class Avatar extends FoundationElement {
+    /**
+     * Indicates the Avatar should have a name.
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: name
+     */
+    @attr public name: string;
+    protected nameChanged(): void {
+        if (!this.initials) {
+            const result = this.name?.match(/\b\w/g) || [];
```

Comment:
I think you need to track a clean/dirty state here. As is, if no initials are set, then name changes (still no initials explicitly set by author) then the initials property won't update with the name change.

I also wonder if this violates the "keep content in light DOM" guiding principle a bit too much. Should the the avatar just expect text content instead of having competing initials & name properties?